lib/utils.js

```diff
@@ -28,5 +28,33 @@ exports.findPrefix = function(xmlnsMapping, nsURI) {
   }
 };
 
-
-
+exports.sign = function(signers, envelope, options, callback) {
+  var next, handled;
+  signers = signers.slice();
+  function returnCallback(error, envelope){
+    if(handled) {
+      return;
+    }
+    handled = true;
+    callback(error,envelope);
+  }
+  next = function(error, envelope) {
+    var signer;
+    if(handled) {
+      return;
+    }
+    if (error) {
+      return returnCallback(error, envelope);
+    }
+    signer = signers.shift();
+    if(!signer) {
+      return returnCallback(null, envelope);
+    }
+    try{
+      signer(envelope, options, next);
+    }catch(error){
+      returnCallback(error,envelope);
+    }
+  };
+  next(null, envelope);
```

Comment:
Can you inline the contents of `next` instead of having a variable and calling it?